src/users/usersActions.js

```diff
@@ -25,17 +26,41 @@ export const reportPresence = (isActive: boolean = true, newUserInput: boolean =
   });
 };
 
-const typingWorker = auth => ({
-  get_current_time: () => new Date().getTime(),
+const typingWorker = (state: GlobalState) => {
+  const auth: Auth = getAuth(state);
 
-  notify_server_start: (user_ids_array: number[]) => {
-    api.typing(auth, JSON.stringify(user_ids_array), 'start');
-  },
+  const getRecipients = user_ids_array => {
+    // User ID arrays are only supported in server versions >= 2.0.0-rc1, for
+    // versions before this, email arrays are used. If current server version is
+    // undetermined, user ID arrays are optimistically used.
```

Comment:
This is... probably the right decision. The server started providing its version in `1.5.0-547-gd9b10727fa`, in early 2017; it's more likely that we failed to parse a new, exotic version-string format than that someone's actually trying to connect to a Zulip server that doesn't provide one.

It may be worth making a note of that reasoning somewhere, though. (Or of yours, if it was different.)